core/controllers/acl_decorators.py

```diff
@@ -2615,3 +2617,153 @@ def test_can_play_entity(self, entity_type, entity_id, **kwargs):
     test_can_play_entity.__wrapped__ = True
 
     return test_can_play_entity
+
+
+def can_access_voiceover_applications(handler):
+    """Decorator to check whether the user can access voiceover applications for
+    a given purpose.
+
+    Args:
+        handler: function. The function to be decorated.
+
+    Returns:
+        function. The newly decorated function that now also checks if a
+            user has permission to access voiceover application for a given
+            purpose.
+    """
+
+    def test_can_access_voiceover_application(self, purpose, **kwargs):
+        """Checks if the user can access voiceover application for a given
+        purpose.
+
+        Returns:
+            *. The return value of the decorated function.
+
+        Raises:
+            NotLoggedInException: The user is not logged in.
+            UnauthorizedUserException: The user does not have
+                credentials to voiceover an exploration.
+        """
+        if not self.user_id:
+            raise base.UserFacingExceptions.NotLoggedInException
+
+        if purpose == feconf.VOICEOVER_APPLICATION_REVIEW:
+            user_actions_info = user_services.UserActionsInfo(self.user_id)
+            if (
+                    role_services.ACTION_ACCEPT_ANY_VOICEOVER_APPLICATION not in
+                    user_actions_info.actions):
+                raise self.UnauthorizedUserException(
+                    'You do not have credentials to review voiceover '
```

Comment:
Why is this message the same as for line 2694? I'm a bit unclear about the difference between the two handlers -- it might be worth clarifying this in the naming/docstrings/messages.

Reply:
(These changes are addressed in the handler PR at #7936, this PR focuses on frontend changes which as detailed in the PR description.)